test/src/service-builders/testAppiumServer.js

```diff
@@ -0,0 +1,356 @@
+const assert = require('assert');
+const mockery = require('mockery');
+const path = require('path');
+const common = require('../../common.js');
+const NightwatchClient = common.require('index.js');
+const Settings = common.require('settings/settings.js');
+
+describe('AppiumServer Transport Tests', function () {
+  beforeEach(function() {
+    mockery.enable({useCleanCache: true, warnOnReplace: false, warnOnUnregistered: false});
+  });
+
+  afterEach(function(){
+    mockery.deregisterAll();
+    mockery.disable();
+    mockery.resetCache();
+  });
+
+  const sessionData = {
+    sessionId: '111',
+    getId() {
+      return '1111';
+    },
+    getCapabilities() {
+      return {
+        getPlatform() {
+          return 'MAC';
+        },
+        getBrowserName() {
+          return 'chrome';
+        },
+        getBrowserVersion() {
+          return '11';
+        },
+        get(name) {
+          return name;
+        },
+        keys() {
+          return new Map();
+        }
+      };
+    }
+  };
+
+  const mockDriver = {
+    session_: Promise.resolve(),
+    async getSession() {
+      return sessionData;
+    },
+    async getExecutor() {
+      return {
+        w3c: true
+      };
+    }
+  };
+
+  const fn = function() {};
+  function deleteFromRequireCache(location) {
+    const entry = Object.keys(require.cache).filter(item => {
+      return item.includes(location);
+    });
+
+    entry.forEach(item => {
+      delete require.cache[item];
+    });
+  }
+
+  function mockHttpClient({onServerUrl = fn}) {
+    function httpClient() {
+      return class HttpClient {
+        constructor(serverUrl) {
+          onServerUrl(serverUrl);
+        }
+      };
+    }
+
+    mockery.registerMock('./httpclient.js', httpClient);
+  }
+
+  function mockExecutor(results) {
+    const {Executor, HttpClient} = require('selenium-webdriver/http');
+    class MockExecutor extends Executor {
+      async execute(command) {
+        return Promise.resolve(results);
+      }
+    }
+
+    deleteFromRequireCache('node_modules/selenium-webdriver/http');
+
+    mockery.registerMock('selenium-webdriver/http', {
+      Executor: MockExecutor,
+      HttpClient
+    });
+  }
```

Comment:
are we using this function?

Reply:
It was being used earlier but now it's not. I'll remove it.

Reply:
Done